src/shared/config/EnvironmentAuthConfig.php

```diff
@@ -0,0 +1,47 @@
+<?php declare(strict_types = 1);
+namespace PharIo\Phive;
+
+class EnvironmentAuthConfig implements AuthConfig {
+    private const KNOWN_TOKEN = [
+        'api.github.com' => 'GITHUB_AUTH_TOKEN',
+        'gitlab.com'     => 'GITLAB_AUTH_TOKEN',
+    ];
+
+    /** @var Environment */
+    private $environment;
+
+    public function __construct(Environment $environment) {
+        $this->environment = $environment;
+    }
+
+    public function hasAuthentication(string $domain): bool {
+        if (!\array_key_exists($domain, self::KNOWN_TOKEN)) {
+            return false;
+        }
+
+        return $this->environment->hasVariable(self::KNOWN_TOKEN[$domain]);
+    }
+
+    /**
+     * @throws AuthException
+     * @throws \BadMethodCallException
+     */
+    public function getAuthentication(string $domain): Authentication {
+        if (!$this->hasAuthentication($domain)) {
+            throw new AuthException(\sprintf('No authentication data for %s', $domain));
+        }
+
+        $token = $this->environment->getVariable(self::KNOWN_TOKEN[$domain]);
+
+        switch ($domain) {
+            // "Token" Authorizations
+            case 'api.github.com':
+                return new Authentication($domain, 'Token', $token);
```

Comment:
What do you think about having explicit implementations of `Authentication` - like `TokenAuthentication` and `BearerAuthentication` (and `BasicAuthentication`) rather than having a generic one with a string parameter? :)

Reply:
The HTTP RFCs about `Authorization` header doesn't restrict the type of credentials (for example, `Token` is not declared anywhere in RFCs, only [`Basic`, `Digest`](https://tools.ietf.org/html/rfc2617) and [`Bearer`](https://tools.ietf.org/html/rfc6750) are).

Another type of credential not represented here is the [AWS S3 one](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-auth-using-authorization-header.html), which goes with the type `AWS4-HMAC-SHA256`

Reply:
But we're not really supporting the other types, or are we?

Technically the current implementation of `Authorization` does not restrict the type in any form, but the public [sig] constants only represent the 'supported' types, the XSD restricts the xml configuration as well.

While I totally agree there may be various additional types possible, we currently do not really have support for them. It might work rather by accident - as long as we do not verify the xml file against the schema.

I'm in favor of making things explict, as you might have noticed already ;). 

So I'd rather explicitly write code to support the Amazon type, for instance, if need be. In particular since the AWS Auth Header seems to be comprised of various sub elements that I envision to be provided individually for readability. Others may even require runtime calculations to be included - like OTP Tokens.

Reply:
Even your AWS Token seems to require runtime calculation for signatures. I only browsed over the Documentation, though...